### DISCUSSION TITLE
Unable to get gtkwave to work - Switch module fix not working

### DISCUSSION THREAD

**Discussion by AetherDim** [Homebrew/discussions#2367] [Casks]
Hey there,

I need gtkwave to look at my simulation results but it just won't work :(

This tutorial does not work for me https://ughe.github.io/2018/11/06/gtkwave-osx
Switch is in both perl directories.

It is not possible to load a file manually:
![image](https://user-images.githubusercontent.com/74829006/139214381-5725e4ca-4080-4e0a-86a1-bb08b0a4e69a.png)

When I load the file using "/Applications/gtkwave.app/Contents/Resources/bin/gtkwave filename.ghw" I get the following:
Unable to block on application (GetProcessPID() returned 18446744073709551016)

If I use the gtkwave command provided by homebrew I get the error mentioned in the tutorial:
find: /share/locale: No such file or directory
/usr/local/bin/gtkwave: line 215: /usr/local/bin/../../../Contents/Resources/bin/: No such file or directory
/usr/local/bin/gtkwave: line 215: exec: /usr/local/bin/../../../Contents/Resources/bin/: cannot execute: No such file or directory

I use zsh.

Anny suggestion would be helpful.

Thanks in advance.

**Comment by carlocab** [comment]
It's relatively unlikely that someone here knows how to help, I'm afraid. You're probably better off asking at the GTKWave mailing list, which you can find at https://sourceforge.net/projects/gtkwave/.

**Reply by AetherDim** [reply]
Thanks for this information.

Will the package be removed from Homebrew?
Because the probability is high that the cask is already broken after installation.

![image](https://user-images.githubusercontent.com/74829006/139217969-5f2d9ef9-3cd1-4e47-953f-b9acd8e79ab0.png)

**Reply by carlocab** [reply]
If someone experienced with the software tells us it's broken, then we'll probably remove it. However, it seems unlikely to me that it is broken. This cask has a few hundred installs a month, and this is the first I'm hearing of problems with it.

**Reply by AetherDim** [reply]
I think it would be nice to install all required dependencies for the app and not hope that Switch is installed.
Is the mentality for casks different?

**Reply by carlocab** [reply]
It's not different -- I agree with you that required dependencies should be installed where we're able to install them. We'd be happy to review a pull request that implements this for the GTKWave cask.

**Reply by AetherDim** [reply]
Thank you very much for your time, I will look into the problem.

**Comment by BlueGreenMagick** [comment]
I also could not get it to work fully, following the guide. 

But `open -a gtkwave dump.vcd` works for me at least.

**Reply by AetherDim** [reply]
A easy solution would probably to alias ```open -a gtkwave``` with ```gtkwave```.

**Reply by AetherDim** [reply]
Two new solutions:

1. Use the ```gtkwave-bin``` binary with the ```-o <filename>``` option. This will print the full program output and other arguments like ```--help``` are possible. Omitting the ```-o``` will result in an "the document could not be opened" error but it seems to work anyway.
2. Use the gtkwave binary inside the Resources/bin directory. This option requires the perl Switch package to be installed. (```cpan install Switch```)

All these files are inside the app.

<img width="285" alt="image" src="https://user-images.githubusercontent.com/74829006/199917879-1abc5cbe-a04e-444b-9a2e-eada97571fad.png">

**Comment by sachinkum0009** [comment]
Hi, greetings 
There seems to be an issue with the gtkwave after the upgrade to Sonoma 14.

The gtkwave shows message with

> This version of “gtkwave-bin” is not compatible with macOS 14 or later and needs to be updated. Contact the app developer for more information.

Are you also facing this issue?

Any suggestion to fix this? Thanks

**Reply by sachinkum0009** [reply]
Found solution for this issue.
https://github.com/gtkwave/gtkwave/issues/250#issuecomment-1798440219
